data/campaigns/Descent_Into_Darkness/units/Malin_Keshar.cfg

```diff
@@ -0,0 +1,702 @@
+        type=cold
+        [specials]
+            {WEAPON_SPECIAL_MAGICAL}
+        [/specials]
+        range=ranged
+        damage=11
+        number=2
+        icon=attacks/iceball.png
+    [/attack]
+    [attack]
+        name=shadow wave
+        description=_"shadow wave"
+        #textdomain wesnoth-did
+        type=arcane
+        [specials]
+            {WEAPON_SPECIAL_MAGICAL}
+        [/specials]
+        range=ranged
+        damage=8
+        number=2
+        icon=attacks/dark-missile.png
+    [/attack]
+#enddef
+
+#define DARK_MAGE_BASE
+    hitpoints=49
+    advances_to=null
+    [attack]
+        name=short sword
+        #textdomain wesnoth-units
+        description=_ "short sword"
+        icon=attacks/sword-human-short.png
+        type=blade
+        range=melee
+        damage=4
+        number=3
+    [/attack]
+    [attack]
+        name=chill wave
+        description=_"chill wave"
+        type=cold
+        [specials]
+            {WEAPON_SPECIAL_MAGICAL}
+        [/specials]
+        range=ranged
+        damage=13
+        number=2
+        icon=attacks/iceball.png
+    [/attack]
+    [attack]
+        name=shadow wave
+        description=_"shadow wave"
+        #textdomain wesnoth-did
+        type=arcane
+        [specials]
+            {WEAPON_SPECIAL_MAGICAL}
+        [/specials]
+        range=ranged
+        damage=10
+        number=2
+        icon=attacks/dark-missile.png
+    [/attack]
+    {DID_CONSUMPTION}
+#enddef
+
+[unit_type]
+    id=Malin Keshar
+    {APPRENTICE_MAGE_BASE}
+    [base_unit]
+        id=Apprentice Mage
+    [/base_unit]
+    [variation]
+        {APPRENTICE_MAGE_BASE}
+        hide_help=yes
+        variation_id=AM1
+        inherit=yes
+        {AM_MELEE AM2}
+        {AM_RANGED AM2}
+    [/variation]
+    [variation]
+        {APPRENTICE_MAGE_BASE}
+        hide_help=yes
+        variation_id=AM2
+        inherit=yes
+        {AM_MELEE AM3}
+        {AM_RANGED AM3}
+        {MALIN_MELEE_MAGICAL AM3}
+    [/variation]
+    [variation]
+        {APPRENTICE_MAGE_BASE}
+        hide_help=yes
+        variation_id=AM3
+        inherit=yes
+        {AM_MELEE_ADVANCE}
+        {AM_RANGED_ADVANCE}
+    [/variation]
+[/unit_type]
+
+[unit_type]
+    id=MalinNecromancer
+    {APPRENTICE_NECROMANCER_BASE}
+    [base_unit]
+        id=Apprentice Necromancer
+    [/base_unit]
+    experience=0
+    {AUTO_ADVANCE 2 35 39 43 AN1}
+
+    [variation]
+        {APPRENTICE_NECROMANCER_BASE}
+        hide_help=yes
+        variation_id=AN1
+        inherit=yes
+        {AM_MELEE AN2}
+        {AM_RANGED AN2}
+        {MALIN_MELEE_MAGICAL AN2}
+        {MALIN_MELEE_PLAGUE AN2}
+        {MALIN_MELEE_CANALIZE AN2}
+    [/variation]
+    [variation]
+        {APPRENTICE_NECROMANCER_BASE}
+        hide_help=yes
+        variation_id=AN2
+        inherit=yes
+        {AM_MELEE AN3}
+        {MALIN_RANGED AN3}
+        {MALIN_MELEE_MAGICAL AN3}
+        {MALIN_MELEE_PLAGUE AN3}
```

Comment:
this looks like he can basicially get the melee_plague spaeical twice, once on the AN1 once on AN2 which will then just the same special again with no effect.

Reply:
He can't because the AMLA option excludes itself, so once he's gotten it once, it'll never show up again.